cmd/minishift/cmd/start_preflight.go

```diff
@@ -414,3 +423,29 @@ func isMounted(driver drivers.Driver, mountpoint string) (bool, error) {
 
 	return true, nil
 }
+
+// checkIsoUrl checks the Iso url and returns true if the iso file exists
+func checkIsoUrl() bool {
```

Comment:
@anjannath , this function can be make testable with adding argument to it.

```
func checkISOURL(url string) bool {
  
   ...
}

// Call as  checkISOURL(viper.GetString(configCmd.ISOUrl.Name))
```

And then add unit test for it. Probably this could be moved to some util package.

Reply:
is there any other place where this function will be needed? To make it testable we will have to add arguments like? and if we don't call it from `preflightCheckSucceedsOrFails` we will loose the nice error, output like `OK` `FAIL`, i don't know if that's abosolutely required but looks nicer.

Reply:
@anjannath , its only about function `checkISOURL`. 
We should always try to make our code testable.  In its current form, I hardly think of writing unit test for it.

Reply:
@budhrg Okay, i'll look at other unit tests and try to make it tesable. But what did you mean by 
`// Call as  checkISOURL(viper.GetString(configCmd.ISOUrl.Name))` this is for testing or calling in the preflight check?

Reply:
@anjannath ,  We probably need to change the  syntax of `preflightCheckSucceedsOrFails` to pass arguments to `preflightCheckFunc`.

Ignore my comments. I will start new issue to have this discussion.

Reply:
@budhrg Okay, right now there are no unit tests for preflight checks. we need to add those

Reply:
@budhrg I disagree. You never write code specifically to be testable... You make code reusable, and testable comes eight this as it likely gets parameterized and has a result value.

Generally, preflight checks have been harder to test as they are specific to a situation. We only test on KVM in our integration. Mocking is suggested, but it is harder to mock power shell sessions, etc. You test what is deemed necessary... Because at the moment automation will not compensate for the lack of manual testing.

Reply:
I actually think we should move some of these tests to validations.go, as we have duplicated the proxy test, and I also see the need we actually do a preflight check to test this, which means three tests?

And what if we also implement a connect test to the proxy? The reusable code is than more valuable to be unit-tested

Reply:
s/than/then